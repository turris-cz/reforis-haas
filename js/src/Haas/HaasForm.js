/*
 * Copyright (C) 2020-2022 CZ.NIC z.s.p.o. (https://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

import React from "react";
import PropTypes from "prop-types";
import { Switch, TextInput } from "foris";

HAASForm.propTypes = {
    formData: PropTypes.shape({
        enabled: PropTypes.bool,
        token: PropTypes.string,
    }).isRequired,
    setFormValue: PropTypes.func,
    disabled: PropTypes.bool,
    formErrors: PropTypes.shape({
        token: PropTypes.string,
    }),
};

HAASForm.defaultProps = {
    formData: {
        enabled: false,
        token: "",
    },
    formErrors: {},
};

export default function HAASForm({
    formData,
    setFormValue,
    disabled,
    formErrors,
}) {
    return (
        <>
            <h2>{_("Proxy Settings")}</h2>
            <Switch
                checked={formData.enabled}
                label={_("Enable HaaS proxy")}
                onChange={setFormValue((value) => ({
                    enabled: { $set: value },
                }))}
                disabled={disabled}
            />
            {formData.enabled && (
                <TextInput
                    value={formData.token || ""}
                    label={_("HaaS token")}
                    error={formErrors.token}
                    helpText={_(
                        `Token used to match SSH session with device in HaaS proxy.`
                    )}
                    onChange={setFormValue((value) => ({
                        token: { $set: value },
                    }))}
                    disabled={disabled}
                />
            )}
        </>
    );
}
