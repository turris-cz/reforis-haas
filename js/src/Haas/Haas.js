/*
 * Copyright (C) 2020-2021 CZ.NIC z.s.p.o. (https://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

import React from "react";
import { ForisForm } from "foris";

import API_URLs from "../API";
import HaasForm from "./HaasForm";

export default function Haas() {
    return (
        <>
            <h1>{_("Honeypot as a Service")}</h1>
            <p
                dangerouslySetInnerHTML={{
                    __html: _(`Honeypot as a Service (HaaS) is a special \
software which simulates an operating system and allows an attacker to log in \
via SSH or telnet and execute commands or download malware. For more \
information click <a href="https://haas.nic.cz/" target="_blank" rel="noopener \
noreferrer">here<sup><i class="fas fa-external-link-alt link-outside-icon \
fa-xs"></i></sup></a>.`),
                }}
            />
            <ForisForm
                forisConfig={{
                    endpoint: API_URLs.haas,
                }}
                validator={validator}
            >
                <HaasForm />
            </ForisForm>
        </>
    );
}

function validator(formData) {
    const error = {};
    const regex = /[a-f0-9]{32,48}/;

    if (formData.token.length > 48 || !regex.test(formData.token))
        error.token = _("Please enter a valid alphanumeric token.");

    return error.token ? error : undefined;
}
