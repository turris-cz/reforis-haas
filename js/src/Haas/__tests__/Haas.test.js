/*
 * Copyright (C) 2020 CZ.NIC z.s.p.o. (http://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

import React from "react";
import mockAxios from "jest-mock-axios";
import { render } from "foris/testUtils/customTestRender";

import Haas from "../Haas";

describe("<Haas />", () => {
    it("should render component", () => {
        const { getByText } = render(<Haas />);
        expect(getByText("Haas")).toBeDefined();
        expect(mockAxios.get).toBeCalledWith(
            "/reforis/haas/api/example",
            expect.anything()
        );
    });
});
