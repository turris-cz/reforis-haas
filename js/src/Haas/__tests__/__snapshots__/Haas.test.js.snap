// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Haas /> Should render HaaS component and accept a valid token. 1`] = `
<div>
  <h1>
    Honeypot as a Service
  </h1>
  <p>
    Honeypot as a Service (HaaS) is a special software which simulates an operating system and allows an attacker to log in via SSH or telnet and execute commands or download malware.
  </p>
  <p>
    If you want to use Honeypot as a Service on your router, you should have the public IPv4 address, register on the HaaS website, and fill out the token. Also, there you can see sessions and statistics from the SSH honeypot from your router. For more information click 
    <a
      href="https://haas.nic.cz/"
      rel="noopener noreferrer"
      target="_blank"
    >
      here
      <sup>
        <i
          class="fas fa-external-link-alt link-outside-icon fa-xs ml-1"
        />
      </sup>
    </a>
    .
  </p>
  <div
    class="card p-4 col-sm-12 col-lg-12 p-0 mb-4"
  >
    <form>
      <h2>
        Proxy Settings
      </h2>
      <div
        class="form-check form-switch mb-3 null"
      >
        <input
          class="form-check-input"
          id="1"
          role="switch"
          type="checkbox"
        />
        <label
          class="form-check-label"
          for="1"
        >
          Enable HaaS proxy
        </label>
      </div>
      <div
        class="mb-3"
      >
        <label
          class="form-label"
          for="2"
        >
          HaaS token
        </label>
        <div
          class="input-group"
        >
          <input
            class="form-control"
            id="2"
            type="text"
            value="0008033cd3b4566ead6c50862d1e0cc7111111abcadf1111"
          />
        </div>
        <div
          class="form-text"
        >
          <small>
            Token used to match SSH session with device in HaaS proxy.
          </small>
        </div>
      </div>
      <div
        class="text-end"
      >
        <button
          class="btn btn-primary col-12 col-md-3 col-lg-2 d-inline-flex justify-content-center align-items-center"
          type="submit"
        >
          <span>
            Save
          </span>
        </button>
      </div>
    </form>
  </div>
</div>
`;

exports[`<Haas /> Should render HaaS component. 1`] = `
<div>
  <h1>
    Honeypot as a Service
  </h1>
  <p>
    Honeypot as a Service (HaaS) is a special software which simulates an operating system and allows an attacker to log in via SSH or telnet and execute commands or download malware.
  </p>
  <p>
    If you want to use Honeypot as a Service on your router, you should have the public IPv4 address, register on the HaaS website, and fill out the token. Also, there you can see sessions and statistics from the SSH honeypot from your router. For more information click 
    <a
      href="https://haas.nic.cz/"
      rel="noopener noreferrer"
      target="_blank"
    >
      here
      <sup>
        <i
          class="fas fa-external-link-alt link-outside-icon fa-xs ml-1"
        />
      </sup>
    </a>
    .
  </p>
  <div
    class="card p-4 col-sm-12 col-lg-12 p-0 mb-4"
  >
    <form>
      <h2>
        Proxy Settings
      </h2>
      <div
        class="form-check form-switch mb-3 null"
      >
        <input
          class="form-check-input"
          id="1"
          role="switch"
          type="checkbox"
        />
        <label
          class="form-check-label"
          for="1"
        >
          Enable HaaS proxy
        </label>
      </div>
      <div
        class="text-end"
      >
        <button
          class="btn btn-primary col-12 col-md-3 col-lg-2 d-inline-flex justify-content-center align-items-center"
          disabled=""
          type="submit"
        >
          <span>
            Save
          </span>
        </button>
      </div>
    </form>
  </div>
</div>
`;
